src/view/dataElements/xdmObject/helpers/generateTreeStructure.js

```diff
@@ -104,38 +89,43 @@ const getTreeNode = ({
     confirmTouchedAtCurrentOrDescendantNode();
   }
 
-  if (
-    !isAncestorUsingWholePopulationStrategy &&
-    isUsingWholePopulationStrategy &&
-    isFormStateValuePopulated(value)
-  ) {
-    confirmDataPopulatedAtCurrentOrDescendantNode();
-  }
-
-  // Type specific helpers should set:
-  //  - children, if the node has children
   getTypeSpecificHelpers(schema.type).populateTreeNode({
     treeNode,
     formStateNode,
+    treeNodeComponent,
     isAncestorUsingWholePopulationStrategy,
-    isUsingWholePopulationStrategy,
-    confirmDataPopulatedAtCurrentOrDescendantNode,
+    isCurrentNodeTheHighestNodeUsingWholePopulationStrategy,
+    doesHighestAncestorWithWholePopulationStrategyHaveAValue,
```

Comment:
I think `isCurrentNodeTheHighestNodeUsingWholePopulationStrategy` and `doesHighestAncestorWithWholePopulationStrategyHaveAValue` can be safely removed. They don't seem to be used anymore.